middleware/verifyUser.js

```diff
@@ -1,39 +1,47 @@
+import { compareSync } from 'bcrypt';
 import { User } from '../models';
 
+let userWantsToLogin = false;
+
 export default (req, res, next) => {
-  const {
-    body,
-    user,
-  } = req;
+  const { body, user } = req;
   const options = {};
   if (user) {
     options.fieldName = 'id';
     options.fieldValue = user.id;
   } else if (body) {
     options.fieldName = 'email';
     options.fieldValue = body.email;
+    userWantsToLogin = true;
   }
 
   const { fieldName, fieldValue } = options;
-  User.findOne({ where: { [fieldName]: fieldValue } })
-    .then((foundUser) => {
-      if (foundUser) {
-        const { dataValues: { verified } } = foundUser;
-        if (!verified) {
-          return res.status(403).json({
+  User.findOne({ where: { [fieldName]: fieldValue } }).then((foundUser) => {
+    if (foundUser) {
+      const {
+        dataValues: { verified, password }
+      } = foundUser;
+      if (!verified) {
+        return res.status(403).json({
+          success: false,
+          errors: ['User has not been verified.']
+        });
+      }
+      if (userWantsToLogin) {
+        const passwordMatch = compareSync(body.password, password);
```

Comment:
@chingsley this logic should not be in a middleware. It should be in the controller. You should be comparing the supplied password in the controller, hence removing the need for your `userWantsToLogin` logic

Reply:
Thanks, @9jaswag. I have made the changes. The password is not compared in the controller.